### PR TITLE
Janitor QoL

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes the janicart use a radial menu rather than a browser for item retrieval, allows for quick mop retrieval with Alt-Click, gives the janicart a more detailed description on examine and allows for the Pimping Ride/Janibot/Xenomaid to toggle their automatic floor cleaning on movement.

## Why It's Good For The Game
This is honestly a pretty big quality of life change as it makes playing Janitor more smooth without needing to rely on the clunky browser menu. This also makes one of the roles that's commonly recommended to new players a lot more accessible.
And the floor autoclean toggle allows for people to pass over floor art/graffiti without erasing it.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
Detailed examine

image

Mop 'quickdraw' with alt click

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/071abb4b-70bc-48f1-8c6a-c0c4fd2ffe14

New janicart item retrieval radial menu

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/4e26cee5-5cd6-4a4f-b079-2b28f42e02d2

Janicart floor buffer toggle

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/5fc4842a-1dc4-4cf1-b14a-fd427df7216d

Janibot floor autoclean toggle

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/e7cbd702-8997-424a-b6e6-08e8abda742a

Xenomaid autopolish toggle

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/cac0e62d-69b4-42a5-a70f-22768780513b

</details>

## Changelog
:cl:
add: Added a more detailed description to the Janicart
add: Added the ability to quickdraw a mop from the Janicart
add: Replaced the old windows menu for the janicart with a radial menu
add: Added the ability to toggle auto floor cleaning for pimping rides, janibots and xeno maids
fix: Fixed the advanced mop having the same inhand icon as the normal mop
/:cl:

### CODE REVIEW

**Comment by Rustyfoxb** [comment]
At last!, I almost never used the cart for mainly this reason, also floor toggle is very nice.

**Comment by Eric-01** [comment]
Since you are touching the janitorial stuff, could you also tweak the advanced mop sprite so that when someone holds it in their hand, you can immediately see which mop it is?

If you don't feel up to it, it could be as simple as keeping the existing sprite and just modifying the hue/saturation/brightness of the mop head to match the advanced mop HUD texture as much as possible. Not perfect, but still better than what we currently have on Bee.

**Comment by CydiaLamiales** [comment]
> Since you are touching the janitorial stuff, could you also tweak the advanced mop sprite so that when someone holds it in their hand, you can immediately see which mop it is?

It already has a different inhand sprite
image

**Comment by Eric-01** [comment]
I didn't know that. However, even if separate sprites already exist in game files, they don't seem to work as expected. I just checked it out to make sure.

Here's me, on the left holding the regular mop and on the right the advanced one:
mop-sprites-in-game

File: code/datums/elements/cleaning.dm [hunk 1]
```diff
@@ -40,3 +39,37 @@
...
 				cleaned_human.regenerate_icons()
 				to_chat(cleaned_human, "<span class='danger'>[AM] cleans your face!</span>")
+
+/datum/action/cleaning_toggle
+	name = "Floor Cleaning Toggle"
+	desc = "Toggles the automatic floor cleaning"
+	icon_icon = 'icons/obj/vehicles.dmi'
+	button_icon_state = "upgrade"
+	var/toggled = TRUE
+	var/atom/movable/toggle_target = null
```
Comment (Penwin0) [inline code/datums/elements/cleaning.dm:49]: This variable isn't getting Garbage collected when the datum is destroyed

**Review by Penwin0 — DISMISSED** [review]

File: code/datums/elements/cleaning.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/datums/elements/cleaning.dm:49]: Resolved

**Review by itsmeow — APPROVED** [review]
Amazing, bye bye Topic()!

**Pull request merged**